### ISSUE TITLE
False positives for Lint/UnusedBlockArgument when using parser_prism

### ISSUE THREAD

**Issue by Darhazer** [rubocop/rubocop#12757]
When using `parser_prism`, `Lint/UnusedBlockArgument` doesn't recognise block argument being used when short hash syntax is used.

```ruby
[1,2,3].map { |x| foo(x:) }
```

The code above does not produce offence with `parser_whitequark` but with `parser_prism` says `Lint/UnusedBlockArgument: Unused block argument - x. You can omit the argument if you don't care about it.`

## RuboCop version

1.62.0 (using Prism 0.24.0, rubocop-ast 1.31.1, running on ruby 3.3.0) [arm64-darwin23]
configured as

```yaml
AllCops:
    ParserEngine: parser_prism
    TargetRubyVersion: 3.3
```

**Comment by bbatsov** [comment]
Most likely an upstream issue that needs to be reported here https://github.com/ruby/prism as well.

**Comment by viralpraxis** [comment]
Cop `Lint/UnusedMethodArgument` also produces false positives, e.g.

```
app/models/assignment.rb:39:18: W: [Correctable] Lint/UnusedBlockArgument: Unused block argument - user. If it's necessary, use _ or _user as an argument name to indicate that it won't be used. Also consider using a proc without arguments instead of a lambda if you want it to accept any arguments but don't care about them.
  scope :for, ->(user) { joins(:course).where(course: { user: }) }
```

Hope it's also fixed in prism head

**Comment by koic** [comment]
This issue has been resolved in Prism. OTOH, the rubocop.gemspec does not have a runtime dependency on Prism. Therefore, there is nothing that can be resolved from the RuboCop side. Please update Prism when a new version is released.
